### PR TITLE
website: Split installation instructions

### LINKED ISSUE #878: Direct-download instructions should be OS-agnostic and helpful
### Summary

The current direct-download instructions are only provided in detail for MacOS and are not conducive to a smooth first user experience.

### Problem Statement

When visiting the [installation instructions](https://opentofu.org/docs/intro/install), the user is presented with a script that only works on MacOS, not on Linux or Windows. When copy-pasting a script (as many users do), the script will error out, causing a bad user experience. OpenTofu is an OS-agnostic tool and the same should apply to its installation instructions.

Furthermore, the current script leaves garbage (ZIP file, LICENSE file, etc) in the user's current working directory, which is not ideal.

### User-facing description

- The user should have a smooth experience getting OpenTofu to run as a direct download.
- The user should not be asked to run a script downloaded from the Internet without being presented with the instructions they are running. (Asking the user to review the script is not really effective.)

### Technical Description

An overhauled script is provided [in this PR](https://github.com/opentofu/opentofu/pull/868). However, a longer script in this section is apparently not desirable.

I personally don't feel comfortable sending in a curlpipe script that downloads a script off the Internet as this is widely considered not best practice.

### Rationale and alternatives

OpenTofu and its predecessor Terraform are and always have been a single binary, which makes it very easy to run. Installing it via a package manager is not necessary. Even in a corporate environment with very restrictive settings on Windows this tool is easily runnable (personal experience). To increase wider adoption, we should make installing OpenTofu as easy as possible and guide first-time users as much as possible. The best possible way is to provide an easy install script, whereas we should provide a script along the best practices. We could take inspiration from [Docker's installation guide](https://docs.docker.com/engine/install/ubuntu/), which provides the easy install script, but still details the steps to get it running.

Alternatively, we could do away with the script example entirely and provide a better guide do download the files from the GitHub releases page. (Possibly with screenshots.)

Another method of providing more instructions would be splitting the page into multiple subpages detailing each installation method. This would also solve the problem of not providing detailed installations for Ubuntu, Debian, RHEL, Fedora, etc. which may be desirable.

### Downsides

Adding a longer script makes the page longer, which, given the multiple installation methods, is apparently not desirable.

### Unresolved Questions

~How do you build the website? There are currently no build instructions provided and I couldn't figure out how to get it running locally.~

### Related Issues

#868

### Proof of Concept

~#868~
#887

### LINKED ISSUE #879: Detailed installation instructions for Linux packages
### Summary

OpenTofu should provide detailed installation instructions instead of a script for OS packages.

### Problem Statement

Currently, the [installation instructions](https://opentofu.org/docs/intro/install) page points to the Packagecloud installation script and asks users to review it. This script is over 200 lines long, is not signed and can change at any time, which would require users to review this script every time they install OpenTofu. There are currently no instructions available for security-concious users who still want to use OS packages.

### User-facing description

[Docker provides a good example](https://docs.docker.com/engine/install/ubuntu/) for installation instructions done right by providing an easy install script, but also detailing the instructions. This makes sure that security-concious users can install the OS package without a need for a curlpipe script.

### Technical Description

The installation instructions page should be broken into separate subpages and installation instructions should be provided separately for each OS variant. (I'm happy to contribute these instructions.)

### Rationale and alternatives

While not everybody has a strong opinion on running untrusted and unverified shell scripts from the Internet, some users do and it is widely shunned as a means of installation. The OpenTofu project should provide the best it can to foster user trust by providing transparent installation instructions that are explained each step of the way.

Alternatively, the Packagecloud script could be shortened down enough to not need an easy install script and can be posted on the website directly.

### Downsides

The installation instructions will need to stay the same over time. If they change, the instructions need to be updated.

### Unresolved Questions

~There are no build instructions for the website, so I can't contribute a tested PoC.~

### Related Issues

#878 
#860

### Proof of Concept

~Happy to contribute one if needed.~
#887

### PR DESCRIPTION
**Pull request by janosdebugs**
This change splits the installation instructions into separate pages and provides manual installation instructions for the OS packages.

Resolves #878
Resolves #879

## Preview

Screenshot of the OpenTofu website, detailing the various installation methods on the left in menu as well as in the main page.

## TODO

- [X] Alpine Linux instructions
- [x] Alpine Linux instructions: Where's the APK signing key?
- [X] DEB instructions
- [X] DEB instructions: Test on Debian
- [X] DEB instructions: Test on Ubuntu
- [X] DEB instructions: Test on Raspbian
- [x] RPM instructions
- [x] RPM instructions: Test on ~RHEL~CentOS Stream
- [x] RPM instructions: Test on Fedora
- [x] RPM instructions: Test on openSUSE
- [X] Snap instructions
- [x] Snap instructions: Test on Ubuntu
- [X] Portable instructions
- [X] Portable instructions: Test on Linux
- [X] Portable instructions: Test on MacOS
- [X] Portable instructions: Test on Windows
- [X] Update installation front page
- [x] [Add the `raw-loader` to package.json on the website repo](https://github.com/opentofu/opentofu.org/pull/213)

### CODE REVIEW

**Comment by janosdebugs** [comment]
Question: is there a published signing key for the APK files on the releases page? I couldn't find any.

**Comment by cam72cam** [comment]
Looking great so far!

**Comment by janosdebugs** [comment]
Thanks @cam72cam ! I'm almost done, one more proof reading session for typos and grammar and a few minor fixed and then it should be good to review.

**Comment by Yantrio** [comment]
>Question: is there a published signing key for the APK files on the releases page? I couldn't find any.

As discussed in slack I think we have used cosign here for the releases in our github repo, but people can feel free to install from the edge repo (until we have a stable release) for alpine by using the following:

```shell
apk add opentofu --repository=http://dl-cdn.alpinelinux.org/alpine/edge/testing/
```

**Comment by janosdebugs** [comment]
On the technical side this PR is ready for review, one more pass on my side for typos and grammar mistakes.

**Review by Yantrio — APPROVED** [review]

**Comment by janosdebugs** [comment]
Grammar/typo pass done.

**Comment by Yantrio** [comment]
Thanks for this, this is a great improvement over what we already had.
Everything looks great to me and I really like your usage of raw-loader here to pull in the scripts file so we can maintain those separately too.

I'll wait for either @Roma36 and/or @damianstasik to approve as they're the gatekeepers of the website right now, and I want to make sure they're happy before we merge.

File: website/docs/intro/install/alpine.mdx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+Install OpenTofu on Alpine Linux.
+---
+
+# Installing OpenTofu on Alpine Linux
+
+OpenTofu is available in the [Alpine Linux testing repository](https://pkgs.alpinelinux.org/packages?name=opentofu) or as an `.apk` package from the [GitHub releases page](https://github.com/opentofu/opentofu/releases/latest/).
+
+## Installing from the testing repository
+
+OpenTofu is available in the Alpine Testing repository and can be installed from there:
```
Comment (kislerdm) [inline website/docs/intro/install/alpine.mdx:14]: **(nitpick, non-blocking)** Hey Jánosh, I find few sentences a bit repetitive, WDYT about the following:

```suggestion
## Installing from the testing repository

OpenTofu is available in the [Alpine Testing repository](https://pkgs.alpinelinux.org/packages?name=opentofu) and can be installed from there:
```

File: website/docs/intro/install/index.mdx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  Learn how to install OpenTofu.
+---
+
+import DocCardList from "@theme/DocCardList";
+
+# Installing OpenTofu
+
+You can install OpenTofu via a wide range of methods. Please select your operating system and installation method:
+
+<DocCardList
```
Comment (kislerdm) [inline website/docs/intro/install/index.mdx:13]: **(praise)** It's a brilliant idea - nicely done! 🎉

File: website/docs/intro/install/macos.mdx [hunk 3]
```diff
@@ -0,0 +1,14 @@
+---
+sidebar_position: 5
+sidebar_label: MacOS
+description: |-
+  Install OpenTofu on MacOS.
+---
+
+# Installing OpenTofu on MacOS
+
+You can use OpenTofu as a [portable binary](portable.mdx) or you can install it using [Homebrew](https://brew.sh/). OpenTofu is available in the Homebrew Core repository, so you can install it by running:
```
Comment (kislerdm) [inline website/docs/intro/install/macos.mdx:10]: **(nitpick)** WDYT about linking the homebrew `tofu` page?

```suggestion
You can use OpenTofu as a [portable binary](portable.mdx) or you can install it using [Homebrew](https://formulae.brew.sh/formula/opentofu). OpenTofu is available in the Homebrew Core repository, so you can install it by running:
```

**Review by kislerdm — APPROVED** [review]
Great job, well done János! Thank you! 🚀 

Apart from a couple of nitpicks, LGTM!

File: website/docs/intro/install/index.mdx (hunk 2 above)
Comment (janosdebugs) [reply website/docs/intro/install/index.mdx:13]: Thank you, glad you like it. The DocCardList stylings really do come in handy for these summary pages, we should do more of those.

File: website/docs/intro/install/alpine.mdx (hunk 1 above)
Comment (kislerdm) [reply website/docs/intro/install/alpine.mdx:14]: Thank you for sharing the illustrations! I didn't think of unfortunate css complexity..

I'd suggest to go with your original proposal as it feels to require least effort for most value 😉

File: website/docs/intro/install/macos.mdx (hunk 3 above)
Comment (janosdebugs) [reply website/docs/intro/install/macos.mdx]: Good catch, done!

File: website/docs/intro/install/index.mdx (hunk 2 above)
Comment (kislerdm) [reply website/docs/intro/install/index.mdx:13]: Definitely!

**Review by kislerdm — APPROVED** [review]

File: website/docs/intro/install/alpine.mdx (hunk 1 above)
Comment (janosdebugs) [reply website/docs/intro/install/alpine.mdx:14]: Sounds good to me, I'll resolve this conversation then. If it ever comes up again, I'll give the admonition styling a do over to make it look better when it has embedded paragraphs. Thanks for the input!

**Review by Roma36 — APPROVED** [review]
Looks good to me, nice improvement!

**Comment by cube2222** [comment]
Thanks a lot for the contribution @janosdebugs!

**Pull request merged**